sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -257,18 +257,24 @@ def _batch_send(
         # Pop it here, so requests doesn't feel bad about additional kwarg
         raise_on_any_failure = kwargs.pop("raise_on_any_failure", True)
         request = self._client._client.post(  # pylint: disable=protected-access
-            url='https://{}/?comp=batch'.format(self.primary_hostname),
+            url='{}://{}/?comp=batch{}{}'.format(
+                self.scheme,
+                self.primary_hostname,
+                kwargs.pop('sas', None),
+                kwargs.pop('timeout', None)
```

Comment:
This is going to format "None" into the URL.....

Reply:
the url looks correct with None....

Reply:
I will make them to ""

Reply:
Query parameters can't just be placed side-by-side - if values are present the need to be joined with '&'

Reply:
& is already appended in delete_blobs

Reply:
in delete blobs we have formated sas into "&thesastoken"
timeout is formated into "&timeout=3"